src/v/storage/log_reader.h

```diff
@@ -259,15 +259,34 @@ class log_reader final : public model::record_batch_reader::impl {
 };
 
 /**
- * Assuming caller has already determined that this batch contains
+ * Assuming caller has already determined that this batch contains (!)
```

Comment:
nit: what is the (!) is referring to?

Reply:
I was trying to draw attention to this line :) Added the word "critical" to it too.